controller/openid_configuration.go

```diff
@@ -0,0 +1,86 @@
+package controller
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/fabric8-services/fabric8-auth/app"
+	"github.com/fabric8-services/fabric8-auth/client"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/jsonapi"
+	"github.com/fabric8-services/fabric8-auth/log"
+	"github.com/fabric8-services/fabric8-auth/rest"
+	"github.com/goadesign/goa"
+)
+
+// OpenidConfigurationController implements the openid_configuration resource.
+type OpenidConfigurationController struct {
+	*goa.Controller
+	configuration LoginConfiguration
+}
+
+// NewOpenidConfigurationController creates a openid_configuration controller.
+func NewOpenidConfigurationController(service *goa.Service, configuration LoginConfiguration) *OpenidConfigurationController {
+	return &OpenidConfigurationController{Controller: service.NewController("OpenidConfigurationController"), configuration: configuration}
+}
+
+// Show runs the show action.
+func (c *OpenidConfigurationController) Show(ctx *app.ShowOpenidConfigurationContext) error {
+
+	// Getting Keycloak's /.well-known/openid-configuration
+	keycloakOpenIDConfigurationEndpoint := c.configuration.GetKeycloakURL() + "/auth/realms/" + c.configuration.GetKeycloakRealm() + "/.well-known/openid-configuration"
+	response, err := http.Get(keycloakOpenIDConfigurationEndpoint)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{}, "request to achieve openid-configuration of keycloak failed")
+		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, err))
+	}
+
+	body, err := ioutil.ReadAll(response.Body)
```

Comment:
It won't be relevant when you remove the call to KC but just FYI. Do not forget to close the response body.